pkg/client/actionclient.go

```diff
@@ -125,6 +134,40 @@ func NewActionClientGetter(acg ActionConfigGetter, opts ...ActionClientGetterOpt
 	return actionClientGetter, nil
 }
 
+type Preflight interface {
+	Run(string, *release.Release) error
+	Name() string
+}
+
+const (
+	PreflightOperationInstall   = "install"
+	PreflightOperationUpgrade   = "upgrade"
+	PreflightOperationUninstall = "uninstall"
+)
+
+type PreflightFunc func(string, *release.Release) error
+
+func NewPreflightFunc(name string, pf PreflightFunc) Preflight {
```

Comment:
Open to naming suggestions here. I'm thinking something like `NewPreflightFromPreflightFunc` is more suitable but feels long

Going to spend some time looking into how other packages tend to do this to get some inspiration but figured I shouldn't hold getting a PR up for review on this.